subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/DaemonJavaCompiler.java

```diff
@@ -68,10 +70,18 @@ protected DaemonForkOptions toDaemonForkOptions(JavaCompileSpec spec) {
         ClassPath compilerClasspath = classPathRegistry.getClassPath("JAVA-COMPILER");
         FlatClassLoaderStructure classLoaderStructure = new FlatClassLoaderStructure(new VisitableURLClassLoader.Spec("compiler", compilerClasspath.getAsURLs()));
 
+        String keepAliveModeStr = System.getProperty(KEEP_DAEMON_ALIVE_PROPERTY, KeepAliveMode.SESSION.name());
+        KeepAliveMode keepAliveMode;
+        try {
+            keepAliveMode = KeepAliveMode.valueOf(keepAliveModeStr);
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException("Invalid value for system property " + KEEP_DAEMON_ALIVE_PROPERTY + ": " + keepAliveModeStr);
```

Comment:
❕ Odd to catch an `IAE` just to rethrow a new `IAE` which doesn't even include the original as the cause.

Reply:
I don't think the original is going to have much more meaning? It will just say `"No enum constant " + enumType.getCanonicalName() + "." + name`, e.g. the resulting exception including this new one would be:
```
java.lang.IllegalArgumentException: Invalid value for system property org.gradle.internal.java.compile.daemon.keepAlive: yourValue
  [stack]
Caused by: java.lang.IllegalArgumentException: No enum constant org.gradle.workers.internal.KeepAliveMode.yourValue
  [stack]
```
I don't think it's helpful.

Reply:
Your message is more useful.  It's only one line of simple code, but throwing away the root cause just seems like it has little upside (I guess I don't see the value in a shorter stacktrack).  Maybe `KeepAliveMode` changes at some point and puts some sort of useful diagnostic info in the exception it throws?  If this happens we'd never see it.

Reply:
Fair, future JDKs may improve enum error messages.